### PR TITLE
Tag normalization

### PR DESCRIPTION
**Pull request by franksmicro**
This is the Nov 4 change that updates video-api.js to normalize new tags as they are entered by the user.  Note that although this should fix the problem going forward, we will still need to script to merge existing tags (TODO).

### CODE REVIEW

File: server/api/video.api.js [hunk 1]
```diff
@@ -17,22 +17,35 @@ const headersValidation = Joi.object({
...
+module.exports = function (server, mongoose, logger) {
   // Update Video Segments Endpoint
-  (function() {
+  (function () {
     const Log = logger.bind(Chalk.magenta('Update Video Segments'));
 
     Log.note('Generating Update Video Segments Endpoint');
 
-    const updateVideoSegmentsHandler = async function(request, h) {
+    const updateVideoSegmentsHandler = async function (request, h) {
```
Comment (JKHeadley) [inline server/api/video.api.js:27]: Just a general note. Formatting changes are good, however I've found that PRs are much easier to review when the format changes are kept separate from the actual code/logic changes, as it is much clearer what is being changed. I usually leave format changes to separate PRs to help prevent confusion.

File: server/api/video.api.js [hunk 2]
```diff
@@ -17,22 +17,35 @@ const headersValidation = Joi.object({
...
         const Segment = mongoose.model('segment');
 
         const { videoId, segments } = request.payload;
-        const video = (await RestHapi.list({
-          model: 'video',
-          query: { ytId: videoId, $embed: ['segments.tags'] },
-        })).docs[0];
+
+        //Log.debug("Segments length:", segments.length);
+        //Log.debug("Segments before:", segments, segments[0].tags);
```
Comment (JKHeadley) [inline server/api/video.api.js:34]: I'm sure I'm also guilty of this, but I think its a good practice to not commit/push commented code unless necessary. If it is necessary then usually a comment explaining why is appropriate.

File: server/api/video.api.js (hunk 1 above)
Comment (JKHeadley) [reply server/api/video.api.js:27]: We will also need to at some point make sure we are using the same styling/formatting tools. Not a huge priority though.

**Review by JKHeadley — APPROVED** [review]
Good to go!

File: server/api/video.api.js (hunk 1 above)
Comment (franksmicro) [reply server/api/video.api.js:27]: Justin, I believe it was one of the linting tools built into Webstorm that made some formatting changes.  I noticed it had split some lines, and in a couple cases even added parentheses in calls.  Once I noticed this, I disabled this linting going forward.

**Comment by JKHeadley** [comment]
Hi @franksmicro , you're welcome to merge this PR whenever you are ready. In general part of my git flow process is to give the honor of merging up to whoever submitted the PR. However if for any reason you would rather me do it that's totally fine too. Thanks!
